src/main/java/twilightforest/item/ItemTFMagicMap.java

```diff
@@ -94,7 +94,7 @@ public void updateMapData(World world, Entity entity, TFMagicMapData mapData) {
                             boolean var20 = xOffset * xOffset + zOffset * zOffset > (drawSize - 2) * (drawSize - 2);
                             int xDraw2 = (xCenter / scaleFactor + xStep - xSize / 2) * scaleFactor;
                             int zDraw2 = (zCenter / scaleFactor + zStep - zSize / 2) * scaleFactor;
-                            int[] biomeFrequencies = new int[256];
+                            int[] biomeFrequencies = new int[32768];
```

Comment:
This should not be hardcoded and instead should be based on the presence (or lack thereof) of ID extenders.